syntax="proto3";
package weibo;

service Weibo{
  rpc Regist(RegistReq)returns(RegistRes){}
  rpc Login(LoginReq)returns(LoginRes){}
  rpc PublishPost(PublishPostReq)returns(PublishPostRes){}
  rpc GetHot(GetHotPostReq)returns(stream PostRes){}
  rpc GetPostByUserId(GetPostByUserIdReq)returns(stream PostRes){}
  rpc PostFeedByRecommend(PostFeedReq)returns(stream PostRes){}
  rpc PostFeedByFollow(PostFeedReq)returns(stream PostRes){}
//  rpc PublishPost
//  rpc GetMyFollowPostFeed
}

message PublishPostRes{
  bool result_code=1;
}

message PublishPostReq{
  string token=1;
  string postText=2;
}

message RegistReq{
  string user_name=1;
  string password=2;
}

message RegistRes{
  string user_name=1;
  uint32 user_id=2;
  uint32 result_code=3;
}

message LoginReq{
  string user_name=1;
  string password=2;
}

message LoginRes{
  uint32 user_id=1;
  uint32 result_code=2;
  string token=3;
}

message GetHotPostReq{
  uint32 post_max_rank=1;
}

message PostRes{
  uint32 post_id=1;
  string text=2;
  string author_name=3;
  int32 trans_nums=4;
  string publish_time=5;
}

message GetPostByUserIdReq{
  uint32 user_id=1;
}

message PostFeedReq{
  int32 each_flush_post_num=1;
  string token=2;
}